Prologue not compatible with PrefixMapping2

### Version

4.7.0-SNAPSHOT

### What happened?

Preface: As mentioned e.g. at https://issues.apache.org/jira/browse/JENA-2309 we are using a prefix cc dump (~2.9K prefixes)
Also, with the [LinkedSparqlQuery](http://lsq.aksw.org/) project, we are parsing logs of SPARQL queries. Often servers have predefined prefixes which do not appear in the logs so query parsing needs a "fragmented" approach with query string being parsed against a preconfigured set of prefixes. Because we want to parse million of queries we do not want to spend time uselessly copying billions of prefixes (all thousands prefixes for every query) - hence, an approach based on e.g. PrefixMapping2 is just what we want. I am mentioning this in order to avoid @afs  from performing a quick fix that seals off the APIs I need by e.g. changing all methods to always do the copying :sweat_smile:

Unfortunately, the following breaks because Prologue setPrefix calls removeNsPrefix:

```java
PrefixMapping pm = new PrefixMapping2(PrefixMapping.Extended);
Query query = new Query();
query.setPrefixMapping(pm);
QueryFactory.parse(query, "PREFIX rdf: <http://foo.bar/baz/> SELECT * {}", null, Syntax.syntaxARQ);
```

### Relevant output and stacktrace

```shell
Caused by: java.lang.UnsupportedOperationException: PrefixMapping2: prefix 'rdf' in the immutable map
	at org.apache.jena.sparql.util.PrefixMapping2.removeNsPrefix(PrefixMapping2.java:68)
	at org.apache.jena.sparql.core.Prologue.setPrefix(Prologue.java:122)
	at org.apache.jena.sparql.lang.QueryParserBase.setPrefix(QueryParserBase.java:117)
```


### Are you interested in making a pull request?

Maybe

--- a/jena-arq/src/main/java/org/apache/jena/sparql/core/Prologue.java
+++ b/jena-arq/src/main/java/org/apache/jena/sparql/core/Prologue.java
@@ -113,14 +113,6 @@ public class Prologue
     public void setPrefix(String prefix, String expansion)
     {
         try {
-            // Removal may involve regeneration of the reverse mapping
-            // so only do if needed.
-            String oldExpansion = prefixMap.getNsPrefixURI(prefix) ;
-            if ( Objects.equals(oldExpansion, expansion) )
-                return ;
-            if ( oldExpansion != null )
-                prefixMap.removeNsPrefix(prefix) ;
-
             prefixMap.setNsPrefix(prefix, expansion) ;
         } catch (PrefixMapping.IllegalPrefixException ex)
         {

